src/layouts/Resources.jsx

```diff
@@ -1,378 +1,166 @@
+  }, [])
 
-  changeHandler = (event) => {
+  const changeHandler = (event) => {
     const { value } = event.target;
-    this.setState({ newResourceRoomName: value });
+    setNewResourceRoomName(value)
   }
 
-  createResourceRoom = async () => {
+  const createResourceRoom = async () => {
     try {
-      const { match } = this.props;
       const { siteName } = match.params;
-      const { newResourceRoomName } = this.state;
       const params = { resourceRoom: newResourceRoomName };
-      await axios.post(`${process.env.REACT_APP_BACKEND_URL}/sites/${siteName}/resource-room`, params, {
-        withCredentials: true,
-      });
+      await axios.post(`${process.env.REACT_APP_BACKEND_URL}/sites/${siteName}/resource-room`, params);
     } catch (err) {
       console.log(err);
     }
   }
 
-  render() {
-    const {
-      resourceCategories,
-      resourcePages,
-      categoryModalIsActive,
-      settingsIsActive,
-      selectedResourcePage,
-      resourceRoomName,
-      newResourceRoomName,
-      isSelectingFile,
-      isFileStagedForUpload,
-      stagedFileDetails,
-    } = this.state;
-    const { match, location } = this.props;
-    const { siteName } = match.params;
-    return (
-      <>
-        {/* Resource Room exists */}
-        <Header />
-        {/* main bottom section */}
-        <div className={elementStyles.wrapper}>
-          <Sidebar siteName={siteName} currPath={location.pathname} />
-          { settingsIsActive
-            ? (
-              <ComponentSettingsModal
-                modalTitle={"Resource Settings"}
-                siteName={siteName}
-                isNewFile={selectedResourcePage.isNewPost}
-                category={selectedResourcePage.category}
-                fileName={selectedResourcePage.fileName}
-                settingsToggle={this.settingsToggle}
-                type="resource"
-                pageFilenames={_.chain(resourcePages)
-                  .filter({ category: selectedResourcePage.category })
-                  .map((file) => file.fileName)
-                  .value()}
-              />
-            )
-            : null}
-
-          {/* main section starts here */}
-          <div className={contentStyles.mainSection}>
-            <div className={contentStyles.sectionHeader}>
-              <h1 className={contentStyles.sectionTitle}>Resources</h1>
-              { resourceRoomName
-                ? (
-                  <ResourceCategoryModal
+  return (
+    <>
+      <Header />
+      {/* main bottom section */}
+      <div className={elementStyles.wrapper}>
+        <Sidebar siteName={siteName} currPath={location.pathname} />
+        {/* main section starts here */}
+        <div className={contentStyles.mainSection}>
+          {/* Page title */}
+          <div className={contentStyles.sectionHeader}>
+            <h1 className={contentStyles.sectionTitle}>Resources</h1>
+          </div>
+          {
+            isLoading ? 'Loading Resources...'
```

Comment:
To be consistent with the `Workspace` layout, we can have some elements which are painted first (such as the `Resource Categories` title and the `Note: Categories cannot be empty, create a resource first to create a Category.` text, and paint the dynamic elements later, instead of loading them together all at once

Reply:
As discussed, we will leave this as is currently. The reason the behaviour differs from Workspace is because of the ResourceRoom, which causes something different to show up - since resources cannot be created without first providing a ResourceRoom name. However, we can only check the resource room name through an API call, so painting some elements first might be jarring if no resource room exists yet, since a user would see static elements first and suddenly have it disappear to be replaced with the resource room elements.